app/controllers/templates_controller.rb

```diff
@@ -79,13 +79,17 @@ def auto_complete_controller_name
   end
 
   def preview
+    if params[:id]
```

Comment:
How about not having this if condition here, and just running `@template ||= resource_class.new(params[type_name_plural])` ? That'd be a bit cleaner imo.